### PR TITLE
`azure_rm_manageddisk_info`, `azure_rm_manageddisk` - Add support for display/modification of disk performance `tier` 

### PR DESCRIPTION
**Pull request by blongv**
##### SUMMARY

[See issue 1786](https://github.com/ansible-collections/azure/issues/1786) 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME

azure_rm_manageddisk_info: display performance tier
azure_rm_manageddisk: modify performance tier

### CODE REVIEW

File: plugins/modules/azure_rm_manageddisk.py [hunk 1]
```diff
@@ -172,6 +172,11 @@
             - The total throughput (MBps) that will be allowed across all VMs mounting the shared disk as ReadOnly.
             - MBps means millions of bytes per second - MB here uses the ISO notation, of powers of 10.
         type: int
+    tier:
+        description:
+            - Performance tier assigned to this disk.
+            - See U(https://learn.microsoft.com/en-us/azure/virtual-machines/disks-change-performance) for more information about disk performance tiers.
+        type: str
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_manageddisk.py:182]: Missing choices;  Argument 'tier' in argument_spec defines choices as (['', 'P1', 'P2', 'P3', 'P4', 'P6', 'P10', 'P15', 'P20', 'P30', 'P40', 'P50', 'P60', 'P70', 'P80']) 

        
        
        
```

**Comment by Fred-sun** [comment]
Small change request!

File: plugins/modules/azure_rm_manageddisk.py (hunk 1 above)
Comment (blongv) [reply plugins/modules/azure_rm_manageddisk.py:182]: I have added improved description and choices.

File: plugins/modules/azure_rm_manageddisk.py [hunk 2]
```diff
@@ -462,6 +490,10 @@ def __init__(self):
             disk_m_bps_read_write=dict(
                 type='int'
             ),
+            tier=dict(
+                type='str',
+                choices=['', 'P1', 'P2', 'P3', 'P4', 'P6', 'P10', 'P15', 'P20', 'P30', 'P40', 'P50', 'P60', 'P70', 'P80']
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_manageddisk.py]: Does not match the document description options. Please click below to update or update the document. thank you!
```suggestion
                choices=['P1', 'P2', 'P3', 'P4', 'P6', 'P10', 'P15', 'P20', 'P30', 'P40', 'P50', 'P60', 'P70', 'P80']
```

File: plugins/modules/azure_rm_manageddisk.py [hunk 3]
```diff
@@ -172,6 +172,28 @@
             - The total throughput (MBps) that will be allowed across all VMs mounting the shared disk as ReadOnly.
             - MBps means millions of bytes per second - MB here uses the ISO notation, of powers of 10.
         type: int
+    tier:
+        description:
+            - Performance tier assigned to this disk.
+            - Only settable for I(storage_account_type=Premium_LRS) disks.
+            - Allowed values are C(P1), C(P2), C(P3), C(P4), C(P6), C(P10), C(P15), C(P20), C(P30), C(P40), C(P50), C(P60), C(P70), C(P80)
+            - See U(https://learn.microsoft.com/en-us/azure/virtual-machines/disks-change-performance) for more information about disk performance tiers.
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_manageddisk.py:180]:
```suggestion
            - See U(https://learn.microsoft.com/en-us/azure/virtual-machines/disks-change-performance) for more information about disk performance tiers.
            - Does not apply to Ultra disks.
```

**Review by xuzhang3 — DISMISSED** [review]

**Comment by xuzhang3** [comment]
@blongv can you help resolve the file conflict?

**Comment by blongv** [comment]
> @blongv can you help resolve the file conflict?

Done.  I also fixed typo `self.etwork_access_policy` to `self.network_access_policy` from dev branch

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**